### DISCUSSION TITLE
Use Math easing function from Commodetto

### DISCUSSION THREAD

**Discussion by stc1988** [Moddable-OpenSource/moddable#1309] [Q&A] [answered]
Now I'm trying to implement animation with Commodetto and Outline.

[piuAll](https://github.com/Moddable-OpenSource/moddable/blob/cfe4736638f5ff7cdeccbb038561b5d7e8b1348f/modules/piu/All/piuAll.js#L460-L489) provides easing functions, but I don't want to include `manifest_piu.json`.

Is there an any idea to include only easing functions?

**Comment by phoddie** [comment]
We could break out the easing code into a module. Then you can include it separate. Here's [a Gist with that](https://gist.github.com/phoddie/a7b5bcad8c2d324ee25ace4fa58cb271). Is that what you have in mind?

**Reply by stc1988** [reply]
Yes, that's what I intended.

**Reply by phoddie** [reply]
Great. Where do you suggest the module belongs in the repository?

**Reply by stc1988** [reply]
If it can be used generally, I think it's good to place it in the`modules/base`.

**Reply by phoddie** [reply]
Interesting. I don't think of it as **that** general. It is kind of graphics focused. Then it could go in Commodetto, which is has the other low-level graphics modules.

FWIW – Piu's Timeline is like Easing, in that it can be used independently of Piu. It could be interesting with Outlines, actually. ;)

**Reply by stc1988** [reply]
I have no plans to implement it, I think the easing function can also be used to control servos like [ServoEasing](https://github.com/ArminJo/ServoEasing)

**Reply by phoddie** [reply]
OK, let's try that. Making the new module is easy, but it requires some work to remove it from Piu and update various manifests.

FWIW – I've used an easing equation with an LED. So, yes, they can be useful in many places. Because I just need one function and it was simple, I just implemented the equivalent in JavaScript.

**Comment by stc1988** [comment, marked as answer]
I will check https://github.com/Moddable-OpenSource/moddable/commit/09be19b428e39acf26ef2e2d02baea12553e604a.
Thank you.

**Reply by phoddie** [reply]
Thanks for taking a look. The code is straightforward, but making Piu also use the common modules was some work. I hope I didn't break any build...

**Reply by stc1988** [reply]
My stack-chan is animated with dynamism 👍 
https://x.com/stc1988/status/1760649479693455473?s=20

**Reply by phoddie** [reply]
Very nice!
